app/login.jsx

```diff
@@ -0,0 +1,43 @@
+import { useAuth0 } from '@auth0/auth0-react';
+import { Header } from './header';
+import { useLocation, useNavigate } from 'react-router';
+import { useEffect } from 'react';
+
+const LoginPage = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { loginWithRedirect, isAuthenticated } = useAuth0();
+  
+  // Navigate the User to the previous page if they tried to access login page while authenticated
+  useEffect(() => {
+    if (isAuthenticated) {
+      navigate(-1);
+    }
+  }, []);
+
+  const handleCustomerLogin = () => {
+    localStorage.setItem("status", "external");
+    loginWithRedirect({ appState: { returnTo: location.state.currentPath, status: "external" } });
+  }
+
+  const handleAdminLogin = () => {
```

Comment:
this and the function above may be able to just be one singular function where you pass in a parameter to indicate the type of login:

```javascript
const handleLogin = (type) => {
  localStorage.setItem('status', type);
  loginWithRedirect({ appState: { returnTo: location.state.currentPath, status: type } })
}
```